templates/Azure-Blob-Auto-Expiration.json

```diff
@@ -0,0 +1,132 @@
+{
+  "id": "/providers/Microsoft.Logic/galleries/public/templates/666271ff871243e59b70460a9c2056db",
+  "type": "Microsoft.Logic/galleries/templates",
+  "name": "666271ff871243e59b70460a9c2056db",
+  "properties": {
+    "author": {
+      "displayName": "Yuan Zheng"
+    },
+    "categoryNames": [
+      "schedule"
+    ],
+    "description": "Once a day, scan a storage account and expire all blobs older than defined value.",
+    "displayName": "Scheduler - Azure Blob Storage TTL",
+
+    "definition": {
+      "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
+      "actions": {
+        "For_each": {
+          "actions": {
+            "Condition": {
+              "actions": {
+                "Delete_blob": {
+                  "description": "If blob is older than the expiration age, delete it",
+                  "inputs": {
+                    "host": {
+                      "connection": {
+                        "name": "@parameters('$connections')['azureblob']['connectionId']"
+                      }
+                    },
+                    "method": "delete",
+                    "path": "/datasets/default/files/@{encodeURIComponent(encodeURIComponent(items('For_each')?['Id']))}"
+                  },
+                  "runAfter": {},
+                  "type": "ApiConnection"
+                }
+              },
+              "description": "Check LastModified timestamp and whether older than the expiration age variable",
+              "expression": "@less(items('For_each')?['LastModified'], addDays(utcnow(), variables('ExpirationAgeInDays')))",
```

Comment:
Here you're essentially doing a string comparison on two values that represent DateTime, will they always be of the same format? I think it's better to convert them to ticks before doing such comparison.

E.g. `@less(ticks(items('For_each')?['LastModified']), ticks(addDays(utcnow(), variables('ExpirationAgeInDays')))`

Reply:
fixed